### PR TITLE
refactor: update BlocObserver instances to use const constructors

### PR DESCRIPTION
**Pull request by ValentinVignal**
## Status

**READY**

## Breaking Changes

NO

## Description

Follows https://github.com/felangel/bloc/pull/3704 and relates to https://github.com/felangel/bloc/issues/3703

This PR updates all the bloc observers to have and use a const constructor.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by ValentinVignal** [comment]
@felangel I named the PR `refactor(bloc)` but I'm not modifying the bloc package. I'm modifying `bloc_test` `flutter_bloc`, `replay_bloc` and several examples.
What title should I give to the PR?

File: packages/replay_bloc/example/lib/main.dart [hunk 1]
```diff
@@ -3,12 +3,16 @@ import 'package:flutter_bloc/flutter_bloc.dart';
...
+  Bloc.observer = const AppBlocObserver();
   runApp(const App());
 }
 
+/// {@template app_bloc_observer}
 /// Custom [BlocObserver] that observes all bloc and cubit state changes.
+/// {@endtemplate}
 class AppBlocObserver extends BlocObserver {
+  /// {@macro app_bloc_observer}
+  const AppBlocObserver();
```
Comment (felangel) [inline packages/replay_bloc/example/lib/main.dart:15]:
```suggestion
  const AppBlocObserver();

```

File: packages/flutter_bloc/example/lib/main.dart [hunk 2]
```diff
@@ -2,12 +2,18 @@ import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
 void main() {
-  Bloc.observer = AppBlocObserver();
+  Bloc.observer = const AppBlocObserver();
   runApp(const App());
 }
 
+/// {@template app_bloc_observer}
+
```
Comment (felangel) [inline packages/flutter_bloc/example/lib/main.dart:10]:
```suggestion
```

**Comment by ValentinVignal** [comment]
Thank you for the updates @felangel ! Do you need me to do anything else on this PR?

**Review by felangel — APPROVED** [review]

**Pull request merged**